lib/slack/src/domain.ts

```diff
@@ -0,0 +1,13 @@
+import { SlackReply, composeSlackReply as coreComposeSlackReply } from './core';
+import { fetchManager } from './infrastructure';
+
+
+async function composeSlackReply(
+  userName: String,
+  getManager: (userName: String) => Promise<String>
+): Promise<SlackReply> {
+  const manager = await fetchManager(userName)
```

Comment:
do you mean getManager here?
```suggestion
  const manager = await getManager(userName)
```

Reply:
(I suppose this would become more obvious once we start writing tests)

Reply:
Good spot, I've changed this. I had been running the code (as a poor mans way of testing), and it did work, but it was using fetchManager from the infrastructure layer, which is a violation of the onion architecture. I think there are ways to test / enforce architectural constraints, at least in some languages, so it would be interesting to think about that sometime. I guess a pull request template would also work as a quick thing (domain.ts shouldn't import infrastructure.ts and so on)

Reply:
I've seen API layering enforced using eslint rules in the past. Honestly, I think that adding jest tests for core.ts will make it *super* obvious when you're doing it wrong (we can worry about testing the rest later).

Reply:
Yep, core should be pretty obvious, as it shouldn't depend on anything. The other layers take a bit more thinking about (for me at least). Enforcing the layers is probably a next step anyway, but interesting to think about.